This is a table comparing the data analysis performed by each of the methods from the various papers.


| **Method**  | **Research Paper**                                                                                                                                                                                             | **Algorithms Compared**                                   | **How they compare**                                                                                                                                                                                                                                                                                                                                                                        | Category of Evaluation type |
|-------------|----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|-----------------------------------------------------------|---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|-----------------------------|
| SPIEC EASIE | [pcbi-1004226-g006](https://journals.plos.org/ploscompbiol/article?id=10.1371/journal.pcbi.1004226#pcbi-1004226-g006)                                                                                          | SPIEC-EASIE, SparCC, CCREPE                               | **Consistency** between two models by computing Hamming Distance (the difference between the upper triangular part of the two adjacency matrices) between reference and new models.                                                                                                                                                                                                         | External data               |
| COZINE      | [ncbi-PMC7768662](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC7768662/)                                                                                                                                       | COZINE, SPIEC-EASIE, Ising                                | **Stability** of edges (The stability of each edge is defined by the proportion of bootstrap samples where the resulting networks include the edge)                                                                                                                                                                                                                                         | Bootstrap analysis          |
| CCLasso     | [academic.oup-211784](https://academic.oup.com/bioinformatics/article/31/19/3172/211784)                                                                                                                       | CCLasso, SparCC                                           | **Frobenius Accuracy** with respect to estimating correlation matrix from data using half samples (measured by the Frobenius norm distance between the estimated correlation matrices and a reference correlation matrix).<br>**Reproducibility** (measured by the fraction of the same edges shared for the two steps in the first reference network which only the top 1/4 edges is used) | Sub-sample analysis         |
| SparCC      | [journal.pcbi.1002687](https://journals.plos.org/ploscompbiol/article?id=10.1371/journal.pcbi.1002687)                                                                                                         | SparCC, Pearson                                           | Computing the number of true-positives (TP), false-positives (FP), true-negatives (TN) and false-negatives (FN) detected in the Pearson network by treating the SparCC network as the true one.                                                                                                                                                                                             | External data               |
| REBACCA     | [researchgate-278731609](https://www.researchgate.net/publication/278731609_Investigating_microbial_co-occurrence_patterns_based_on_metagenomic_compositional_data)                                            | REBACCA, SparCC, BP, ReBoot                               | Sensitivity and Specificity (TODO with respect to what?)                                                                                                                                                                                                                                                                                                                                    |                             |
| MDiNE       | [researchgate-337112062](https://www.researchgate.net/publication/337112062_MDiNE_A_model_to_estimate_differential_co-occurrence_networks_in_microbiome_studies)                                               | MDiNE , MInt, COZINE, SPIEC-EASIE                         | The number of edges in a sample. An edge is shown when the co-occurrence between a pair of taxa differs signiﬁcantly between the two groups based on the 90% credible interval.                                                                                                                                                                                                             | Sub-sample analysis         |
| gCoda       | [researchgate-316849033](https://www.researchgate.net/publication/316849033_gCoda_Conditional_Dependence_Network_Inference_for_Compositional_Data)                                                             | gCoda, SPIEC-EASIE                                        | The numbers of **inferred edges** and **edge density** in a sample.                                                                                                                                                                                                                                                                                                                         | Sub-sample analysis         |
| MixMPLN     | [academic.oup-5529264](https://academic.oup.com/bioinformatics/article/35/14/i23/5529264)                                                                                                                      | MixMPLN + glasso, MixMPLN + huge                          | Optimal number of components(taxas) from the sample-taxa matrix is inferred using the [silhouette method](https://rdrr.io/cran/factoextra/man/fviz_silhouette.html) from the [`factoextra`](https://rdrr.io/cran/factoextra/) R package. (Optimal implies highest silhouette width)                                                                                                         | Sub-sample analysis         |
| MINT        | [researchgate-314088430](https://www.researchgate.net/publication/314088430_MINT_A_multivariate_integrative_method_to_identify_reproducible_molecular_signatures_across_independent_experiments_and_platforms) | MINT, PLSDA, sPLSDA, ComBat+sPLSDA, RF, mgPLS, BMC+PLSDA  | **Classification accuracy, Reproducibility and Balanced Error Rates (BER)** with respect to inferring gene signatures from cells.                                                                                                                                                                                                                                                           | External data               |
| mLDM        | [biorxiv-10.1101](https://www.biorxiv.org/content/10.1101/042630v1.full)                                                                                                                                       | mLDM,PCC,SCC,CCREPE, SparCC, CCLasso, glasso, SPIEC-EASIE | Ability to infer OTU-OTU associations(specifically symbiotic interactions like parasitism, mutualism, etc ) from metagenomic sequencing data.                                                                                                                                                                                                                                               | External data               |
| NetCoMi     | [academic.oup-6017455](https://academic.oup.com/bib/article/22/4/bbaa290/6017455)                                                                                                                              | SparCC, SPIEC-EASI, SPRING                                | Variability of microbial networks between the association estimation methods, Aitchison’s distance (Euclidean distance between clr-transformed compositions) between taxas for constructing sample similarity networks.                                                                                                                                                                     | Sub-sample analysis         |
