This is a table comparing the data analysis performed by each of the methods from the various papers.




| **Method**  | **Research Paper**                                                                                                                                                                                             | **Algorithms Compared**                                   | **How they compare**                                                      |
|-------------|----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|-----------------------------------------------------------|---------------------------------------------------------------------------|
| SPIEC EASIE | [pcbi-1004226-g006](https://journals.plos.org/ploscompbiol/article?id=10.1371/journal.pcbi.1004226#pcbi-1004226-g006)                                                                                          | SPIEC-EASIE, SparCC, CCREPE                               | Reproducibility, Consistency, Hamming Distance                            |
| COZINE      | [PMC7768662](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC7768662/)                                                                                                                                            | COZINE, SPIEC-EASIE, Ising                                | Phylo Tree, Assortativity                                                 |
| CCLasso     | [211784](https://academic.oup.com/bioinformatics/article/31/19/3172/211784)                                                                                                                                    | CCLasso, SparCC                                           | Frobenius Accuracy and Reproducibility                                    |
| SparCC      | [journal.pcbi.1002687](https://journals.plos.org/ploscompbiol/article?id=10.1371/journal.pcbi.1002687)                                                                                                         | SparCC, Pearson                                           | Pairwise correlations between all OTUs                                    |
| REBACCA     | [researchgate-278731609](https://www.researchgate.net/publication/278731609_Investigating_microbial_co-occurrence_patterns_based_on_metagenomic_compositional_data)                                            | REBACCA, SparCC, BP, ReBoot                               | Sensitivity and Specificity                                               |
| MDiNE       | [researchgate-337112062](https://www.researchgate.net/publication/337112062_MDiNE_A_model_to_estimate_differential_co-occurrence_networks_in_microbiome_studies)                                               | MDiNE , MInt, COZINE, SPIEC-EASIE                         | The number of edges in a sample                                           |
| gCoda       | [researchgate-316849033](https://www.researchgate.net/publication/316849033_gCoda_Conditional_Dependence_Network_Inference_for_Compositional_Data)                                                             | gCoda, SPIEC-EASIE                                        | The numbers of inferred edges                                             |
| MixMPLN     | [academic.oup-5529264](https://academic.oup.com/bioinformatics/article/35/14/i23/5529264)                                                                                                                      | MixMPLN + glasso, MixMPLN + huge                          | Optimal number of components from taxa matrix.                            |
| MINT        | [researchgate-314088430](https://www.researchgate.net/publication/314088430_MINT_A_multivariate_integrative_method_to_identify_reproducible_molecular_signatures_across_independent_experiments_and_platforms) | MINT, PLSDA, sPLSDA, ComBat+sPLSDA, RF, mgPLS, BMC+PLSDA  | Classification accuracy, The classification<br>Balanced Error Rates (BER) |
| mLDM        | [biorxiv-10.1101](https://www.biorxiv.org/content/10.1101/042630v1.full)                                                                                                                                       | mLDM,PCC,SCC,CCREPE, SparCC, CCLasso, glasso, SPIEC-EASIE | Power of association inference                                            |
| NetCoMi     | [academic.oup-6017455](https://academic.oup.com/bib/article/22/4/bbaa290/6017455)                                                                                                                              | SparCC, SPIEC-EASI, SPRING                                | Variability of microbial networks , Aitchison’s distance                  |

- The **Consistent Accuracy** is measured by the Frobenius norm distance between the estimated correlation matrices of the first and second step.
- The **Consistent reproducibility** is measured by the fraction of the same edges shared for these two steps in the first gold reference network which only the top 1/4 edges is used.
